swarm/src/lib.rs

```diff
@@ -1016,6 +1016,19 @@ where
                     });
                 }
             }
+            ToSwarm::ListenOn { id, address } => {
+                if let Err(e) = self.transport.listen_on(id, address) {
+                    self.behaviour.on_swarm_event(FromSwarm::ListenerError(
+                        behaviour::ListenerError {
+                            listener_id: id,
+                            err: &e,
+                        },
+                    ));
```

Comment:
When a call to `Swarm::listen_on` fails, we return the `TransportError` to the user, but don't call `Behaviour::on_swarm_event`.

https://github.com/libp2p/rust-libp2p/blob/67b26cce4f5563e5b8e45d28731d24f24e468914/swarm/src/lib.rs#L368-L381

When we handle a `ToSwarm::ListenOn` and the call fails, we provide the error to `Behaviour::on_swarm_event` but not to the user.

Would it make sense to always provide the error both to the `NetworkBehaviour` and to the user? That would be in-line with our general goal of mirroring the behavi**o**r from the `Swarm` to the user and the `Swarm` to the `NetworkBehaviour`.

Reply:
Yes I think it would make more sense to provide it to both `NetworkBehaviour` and the user.